### PR TITLE
MultiplyMerge layer added

### PR DESCRIPTION
**Pull request by haritha1313**

### CODE REVIEW

File: src/mlpack/methods/ann/layer/multiply_merge.hpp [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    typename OutputDataType = arma::mat,
+    typename... CustomLayers
+>
+class MultiplyMerge
+{
+ public:
+  /**
+   * Create the MultiplyMerge object using the specified parameters.
+   *
+   * @param model Expose the all network modules.
```
Comment (zoq) [inline src/mlpack/methods/ann/layer/multiply_merge.hpp]: Pedantic style issue, ``Expose all the modules``.

File: src/mlpack/methods/ann/layer/CMakeLists.txt [hunk 2]
```diff
@@ -59,6 +59,8 @@ set(SOURCES
   mean_pooling_impl.hpp
   multiply_constant.hpp
   multiply_constant_impl.hpp
+  multiply_merge.hpp
```
Comment (zoq) [inline src/mlpack/methods/ann/layer/CMakeLists.txt]: Can you add the layer to the list of layer types in: https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/ann/layer/layer_types.hpp

File: src/mlpack/methods/ann/layer/multiply_merge.hpp (hunk 1 above)
Comment (zoq) [reply src/mlpack/methods/ann/layer/multiply_merge.hpp]: Just realized, the ``add_merge`` layer does have the same grammar issue; I can fix this once we merge this one or we could include it here, your choice.

Comment (haritha1313) [reply src/mlpack/methods/ann/layer/multiply_merge.hpp]: I have done the necessary edits in this one itself.

Comment (zoq) [reply src/mlpack/methods/ann/layer/multiply_merge.hpp]: Nice, thanks!

**Pull request merged**